### PR TITLE
Group reverse dependencies by type

### PR DESCRIPTION
**Pull request by zbynek**
The reverse dependencies are currently listed without indication which one is reqired/optional/implied.
Also reverse dependencies are not shown at all if plugin has no dependencies. Both issues are fixed here.

I didn't manage to find a GraphQL query that would work using the JenkinsPlugin type (and would not slow down everything by grabbing too much data), so I aded JenkinsPluginDependency. It's not very DRY but since the data is just used internally it should not matter.

### CODE REVIEW

**Review by markjacksonfishing — APPROVED** [review]

File: plugins/gatsby-source-jenkinsplugins/utils.js [hunk 1]
```diff
@@ -115,6 +115,20 @@ const fetchPluginData = async ({createNode, reporter}) => {
                         contentDigest: crypto.createHash('md5').update(`plugin${pluginData.name.trim()}`).digest('hex')
                     }
                 });
+                pluginData.dependencies.forEach(dependency => {
+                    const mergedId = `${pluginData.name.trim()}:${dependency.name.trim()}`;
+                    createNode({
```
Comment (halkeye) [inline plugins/gatsby-source-jenkinsplugins/utils.js]: so createNode returns a promise (https://www.gatsbyjs.com/docs/actions/#createNode) that means all these dependency nodes won't be waiting until completion.


```suggestions
                createNode().then(pluginNode => Promise.all(pluginData.dependencies.map(dependency => {
                    const mergedId = `${pluginData.name.trim()}:${dependency.name.trim()}`;
                    return createNode({});
                  });
```

**Comment by halkeye** [comment]
Btw, if you want to get an auto buildable for your PR like I have for #484 via digital ocean apps (free for static sites)

UI:
image

Yaml:
[plugin-site.txt](https://github.com/jenkins-infra/plugin-site/files/5620497/plugin-site.txt)

**Review by halkeye — APPROVED** [review]

**Comment by halkeye** [comment]
@zbynek CI is failing:

image

**Comment by halkeye** [comment]
oh createnode mock isn't returning a promise :D

**Pull request merged**